src/Datasets/PluckPreprocessor.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Pest\Datasets;
+
+use InvalidArgumentException;
+use Pest\Contracts\Dataset\Preprocessor;
+
+/**
+ * @internal
+ */
+final class PluckPreprocessor implements Preprocessor
+{
+    public function getKey(): string
+    {
+        return 'pluck';
+    }
+
+    public function process(array $dataset, mixed $value): array
+    {
+        if (!is_string($value)) {
+            throw new InvalidArgumentException('The pluck preprocessor should be given a string.');
+        }
+
+        // @phpstan-ignore-next-line
+        return array_map(fn (array|object $values) => $values[$value], $dataset);
```

Comment:
Won't this still fail on an object because you can't access a standard object property with `[]` syntax. We would have to implement something similar to Laravel's `data_get` or (perhaps simpler) use a ternary to decide how to access the property (`[]` or `->`).

Reply:
Maybe a `data_get` is a bit overkill for this case alone. Do any other uses come in you mind for it to be worth to implement in pest?

Reply:
meanwhile I added a ternary implementation and a test to point out this behavior

it seems fine to me, let me know :wink: